licensing/src/main/java/io/crate/license/SelfGeneratedLicense.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.license;
+
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+
+import java.io.IOException;
+
+import static io.crate.license.DecryptedLicenseData.EXPIRATION_DATE_IN_MS;
+import static io.crate.license.DecryptedLicenseData.ISSUED_TO;
+
+final class SelfGeneratedLicense {
+
+    private SelfGeneratedLicense() {
+    }
+
+    static byte[] encryptLicenseContent(byte[] content) {
+        return Cryptos.encrypt(content);
+    }
+
+    static DecryptedLicenseData decryptLicenseContent(byte[] encryptedContent) throws IOException {
+        byte[] decryptedContent = Cryptos.decrypt(encryptedContent);
+        long expirationDate = 0;
+        String issuedTo = null;
+        try (XContentParser parser = XContentFactory.xContent(XContentType.JSON)
```

Comment:
Having the parser logic here goes a bit against the notion introduced so far that the `DecryptedLicenseData` knows about the structure. Maybe we could move this code as well?